accounts/service.go

```diff
@@ -93,12 +96,15 @@ func (s *InterceptorService) Start(lightningClient lndclient.LightningClient,
 	s.routerClient = routerClient
 	s.checkers = NewAccountChecker(s, params)
 
+	s.isEnabled = true
+
 	// Let's first fill our cache that maps invoices to accounts, which
 	// allows us to credit an account easily once an invoice is settled. We
 	// also track payments that aren't in a final state yet.
 	existingAccounts, err := s.store.Accounts()
 	if err != nil {
-		return fmt.Errorf("error querying existing accounts: %w", err)
+		return s.disableAndErrorf("error querying existing "+
```

Comment:
So unless I'm completely wrong, we can't use the status manager to block gRPC requests from account users, which are making are making lnrpc calls with a macaroon with the account caveat attached to it. This is why the `!s.IsRunning()` check is kept in `accounts/interceptor.go` even after the commit that adds the accounts subsystem to status manager, to block the account users calls when the accounts service has been disabled (either through erroring or through config).
So if we didn't `disableAndErrorf` here, the accounts users gRPC calls would still pass through the `interceptor.go`, and fail later in the processing due to the db being closed or similar.

I might be completely wrong in my understanding here though, so please let me know if this can be done in a better way.

Reply:
oh ok I see 👍  makes sense